### ISSUE TITLE
Error: sys cache is full but there are no clean entries to pop

### ISSUE THREAD

**Issue by ajsauter** [NREL-SIIP/PowerSimulations.jl#834]
As requested, please find my error report below.

**Describe the bug**
Toward the end of my most recent simulation run, I started to see this recurring error message just before the progress bar for the simulation would reappear:

`Barrier solved model in 44 iterations and 12.27 seconds (7.46 work units)
Optimal objective 2.75945856e+07

Crossover log...

    1217 DPushes remaining with DInf 0.0000000e+00                13s
       0 DPushes remaining with DInf 0.0000000e+00                15s

    4457 PPushes remaining with PInf 5.0101744e-05                15s
    2870 PPushes remaining with PInf 5.0101733e-05                15s
       0 PPushes remaining with PInf 5.0101744e-05                16s

  Push phase complete: Pinf 5.0101744e-05, Dinf 1.5249151e-05     16s

Iteration    Objective       Primal Inf.    Dual Inf.      Time
    5677    2.7594586e+07   5.010174e-05   0.000000e+00     16s
    5678    2.7594586e+07   0.000000e+00   0.000000e+00     18s

Solved with barrier
Solved in 5678 iterations and 31.94 seconds (15.01 work units)
Optimal objective  2.759458626e+07

User-callback calls 15520812, time in user-callback 6.44 sec
┌ Error: sys cache is full but there are no clean entries to pop: PowerSimulations.OptimizationResultCacheKey(:UC, PowerSimulations.ParameterKey{ActivePowerTimeSeriesParameter, RenewableDispatch}(""))
└ @ PowerSimulations C:\Users\A.J. Sauter\.julia\packages\PowerSimulations\wNZsa\src\simulation\model_output_cache.jl:68
┌ Error: sys cache is full but there are no clean entries to pop: PowerSimulations.OptimizationResultCacheKey(:UC, PowerSimulations.ParameterKey{RequirementTimeSeriesParameter, VariableReserve{ReserveUp}}("REG_UP"))
└ @ PowerSimulations C:\Users\A.J. Sauter\.julia\packages\PowerSimulations\wNZsa\src\simulation\model_output_cache.jl:68
┌ Error: sys cache is full but there are no clean entries to pop: PowerSimulations.OptimizationResultCacheKey(:UC, PowerSimulations.ParameterKey{RequirementTimeSeriesParameter, VariableReserve{ReserveUp}}("SPIN"))
└ @ PowerSimulations C:\Users\A.J. Sauter\.julia\packages\PowerSimulations\wNZsa\src\simulation\model_output_cache.jl:68
┌ Error: sys cache is full but there are no clean entries to pop: PowerSimulations.OptimizationResultCacheKey(:UC, PowerSimulations.ParameterKey{ActivePowerTimeSeriesParameter, PowerLoad}(""))
└ @ PowerSimulations C:\Users\A.J. Sauter\.julia\packages\PowerSimulations\wNZsa\src\simulation\model_output_cache.jl:68
┌ Error: sys cache is full but there are no clean entries to pop: PowerSimulations.OptimizationResultCacheKey(:UC, PowerSimulations.ParameterKey{RequirementTimeSeriesParameter, VariableReserve{ReserveDown}}("REG_DN"))
└ @ PowerSimulations C:\Users\A.J. Sauter\.julia\packages\PowerSimulations\wNZsa\src\simulation\model_output_cache.jl:68
┌ Error: sys cache is full but there are no clean entries to pop: PowerSimulations.OptimizationResultCacheKey(:UC, PowerSimulations.ParameterKey{ActivePowerTimeSeriesParameter, HydroDispatch}(""))
└ @ PowerSimulations C:\Users\A.J. Sauter\.julia\packages\PowerSimulations\wNZsa\src\simulation\model_output_cache.jl:68
┌ Error: sys cache is full but there are no clean entries to pop: PowerSimulations.OptimizationResultCacheKey(:UC, PowerSimulations.ExpressionKey{PowerSimulations.ProductionCostExpression, RenewableDispatch}(""))
└ @ PowerSimulations C:\Users\A.J. Sauter\.julia\packages\PowerSimulations\wNZsa\src\simulation\model_output_cache.jl:68
┌ Error: sys cache is full but there are no clean entries to pop: PowerSimulations.OptimizationResultCacheKey(:UC, PowerSimulations.ExpressionKey{PowerSimulations.ProductionCostExpression, ThermalMultiStart}(""))
└ @ PowerSimulations C:\Users\A.J. Sauter\.julia\packages\PowerSimulations\wNZsa\src\simulation\model_output_cache.jl:68

Progress: 100%|█████████████████████████████████████████████████| Time: 5 days, 4:05:35
  Step:                  365
  model:                 UC
  Simulation Timestamp:  2018-12-31T00:00:00`

**To Reproduce**
1. Run TAMU ERCOT power system UC&ED problem with the following settings, initial conditions, constraints, etc. using the PSI #master branch:

`using PowerSystems
using PowerSimulations
using InfrastructureSystems
const PSI = PowerSimulations
using CSV
using XLSX
using Plots
using Dates
using PyPlot
using DataFrames
using TimeSeries
using Gurobi #Cbc

local_dir = "C:\\Users\\A.J. Sauter\\Documents"
system = System(joinpath(local_dir, "Local_Sys_Files/tamu_DA_sys_LVLred.json"))
template_uc = ProblemTemplate(NetworkModel(DCPPowerModel, use_slacks = true, duals = [NodalBalanceActiveConstraint]))

set_device_model!(template_uc, ThermalMultiStart, ThermalBasicUnitCommitment)
set_device_model!(template_uc, RenewableDispatch, RenewableFullDispatch)
set_device_model!(template_uc, PowerLoad, StaticPowerLoad)
set_device_model!(template_uc, HydroDispatch, FixedOutput)
set_device_model!(template_uc, Line, StaticBranchUnbounded)
set_device_model!(template_uc, Transformer2W, StaticBranchUnbounded)
set_device_model!(template_uc, TapTransformer, StaticBranchUnbounded)
set_service_model!(template_uc, VariableReserve{ReserveUp}, RangeReserve)
set_service_model!(template_uc, VariableReserve{ReserveDown}, RangeReserve)

models = SimulationModels(
    decision_models = [
        DecisionModel(
            template_uc,
            system;
            name = "UC",
            optimizer = optimizer_with_attributes(
                Gurobi.Optimizer,
                #"logLevel" => 1,
                #"ratioGap" => 0.5
            ),
            system_to_file = false,
            initialize_model = false,
            #calculate_conflict = true,
            optimizer_solve_log_print = true,
            direct_mode_optimizer = true,
        )
    ]
)

DA_sequence = SimulationSequence(models = models, ini_cond_chronology = InterProblemChronology())

sim = Simulation(
    name = string("dwpt-week-", tran_set),
    steps = 365,
    models = models,
    sequence = DA_sequence,
    initial_time = DateTime("2018-01-01T00:00:00"),
    simulation_folder = DATA_DIR,
)
build_out = build!(sim, serialize = false)
execute!(sim)
results = SimulationResults(sim);
uc_results = get_problem_results(results, "UC"); # UC stage result metadata
set_system!(uc_results, system)` 

The two most prominent changes to the system file are the addition of EV loads as additional timeseries files at each load bus and the reduced reactance on all system lines less than 230 kV.

**Expected behavior**
The simulation operated nominally for the majority of the week. The Gorubi solver updates/reporting did not change with the advent of the error message. Noticed the recurring error message on the fifth and final morning of the run. The simulation finished with what appear to be normal variable outputs.

**Comment by jd-lara** [comment]
@ajsauter in Slack you reported this error 


```
 Error: sys cache is full but there are no clean entries to pop: PowerSimulations.OptimizationResultCacheKey(:UC, PowerSimulations.ParameterKey{ActivePowerTimeSeriesParameter, RenewableDispatch}(""))

**Comment by ajsauter** [comment]
@jd-lara yes, the error is a bit further down in the pasted log. Sorry! I wanted to give a bit of context as to where the error was showing up.

**Comment by jd-lara** [comment]
@ajsauter have you seen this error again?

**Comment by ajsauter** [comment]
@jd-lara yes, the error has reproduced for several of my most recent runs. The error was reproduced on a remote desktop session as well as when I was using CU's Research Computing resource, which is similar to NREL's HPC.

**Comment by jd-lara** [comment]
closed via #847
